File: rust/src/models/google_cloud_pub_sub_patch_config.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
pub struct GoogleCloudPubSubPatchConfig {
    #[serde(rename = "projectId")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub project_id: Option<String>,

    #[serde(rename = "topicId")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub topic_id: Option<String>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub credentials: Option<String>,
}

impl GoogleCloudPubSubPatchConfig {
    pub fn new() -> Self {
        Self {
            project_id: None,
            topic_id: None,
            credentials: None,
        }
    }
}

impl Default for GoogleCloudPubSubPatchConfig {
    fn default() -> Self {
        Self::new()
    }
}

File: javascript/src/models/googleCloudPubSubPatchConfig.ts
// this file is @generated

export interface GoogleCloudPubSubPatchConfig {
  projectId?: string;
  topicId?: string;
  credentials?: string;
}

export const GoogleCloudPubSubPatchConfigSerializer = {
  _fromJsonObject(object: any): GoogleCloudPubSubPatchConfig {
    return {
      projectId: object["projectId"],
      topicId: object["topicId"],
      credentials: object["credentials"],
    };
  },

  _toJsonObject(self: GoogleCloudPubSubPatchConfig): any {
    return {
      projectId: self.projectId,
      topicId: self.topicId,
      credentials: self.credentials,
    };
  },
};
